src/Hound.java

```diff
@@ -0,0 +1,83 @@
+
+
+
+public class Hound {
+	private int x_location;
+	private int y_location;
+	private Directions direction;
+	
+	
+	public Hound()
+	{
+		this.x_location =  0;
+		this.y_location = 0;
+		
+	}
+	
+	public Hound(int x, int y)
+	{
+		this.x_location =  x;
+		this.y_location = y;
+	}
+	public void setLocation(int x,int y)
+	{
+		this.x_location =  x;
+		this.y_location = y;
+	}
+	public int getXLocation()
+	{
+		return this.x_location;
+	
+	}
+	public int getYLocation()
+	{
+		return this.y_location;
+	}
+	public void move(int units, Directions direction)
+	{
+		if(direction.getDirection() == 270)
+		{
+			this.y_location += units;
+		}
+		else if(direction.getDirection() == 180)
+		{
+			this.x_location -= units;
+		}
+		else if(direction.getDirection() == 90)
+		{
+			this.y_location -= units;
+		}
+		else if(direction.getDirection() == 0)
+		{
+			this.x_location += units;
+		}
+		
+		else
+		{
+			
+		}
+	}
+	
+	public void setDirection(Directions direction)
+	{
```

Comment:
This whole method is really weird and I can't make any sense out of the logic. setDirection() should not do anything other than set direction to the given value. If you're trying to constrain the Hare to 90 degree turns, there is a much much easier way to do that: just keep calling randomDirection() until you get a valid direction (one that is not opposite the current direction). (You should do this in the caller, not here).